share/spice/clojure_packages_from_clojars/clojure_packages_from_clojars.js

```diff
@@ -0,0 +1,51 @@
+(function(env) {
+    "use strict";
+    env.ddg_spice_clojure_packages_from_clojars = function(api_result) {
+
+        // Validate the response (customize for your Spice)
+        var search_terms = DDG.get_query().split(" ");
+
+        function keep_it(candidate) {
+            return DDG.isRelevant(candidate, ['clojure', 'clojars', 'lib', 'library'], 1, 0);
+        }
+
+        var clojar_artifact = search_terms.filter(keep_it);
+
+        if (!api_result || api_result.error) {
+            return Spice.failed('clojure_packages_from_clojars');
+        }
+
+        // Render the response
+        Spice.add({
+            id: "clojure_packages_from_clojars",
+            // Customize these properties
+            name: "\"" + clojar_artifact + "\"" + ' on Clojars',
+            data: api_result.results,
+
+            meta: {
+                sourceName: "clojars.org",
+                sourceUrl: 'https://clojars.org/search?q=' + clojar_artifact
+            },
+            templates: {
+                group: 'text',
+                detail: false,
+                item_detail: false,
+
+                variants: {
+                    tileTitle: 'basic4',
+                    detail: 'light'
+                }
+            },
+            normalize: function(item) {
+                return {
+                    title: item.group_name + "/" + item.jar_name,
+                    subtitle: item.version,
+                    description: item.description,
```

Comment:
Thank you for your feedback. I'll adress all of this and let you know when
I'm finished.
Le 11 août 2015 22:08, "Zaahir Moolla" <notifications@github.com> a écrit :

> In
> share/spice/clojure_packages_from_clojars/clojure_packages_from_clojars.js
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/2087#discussion_r36799120>
> :
>
> > +            },
> > +            templates: {
> > +                group: 'text',
> > +                detail: false,
> > +                item_detail: false,
> > +
> > +                variants: {
> > +                    tileTitle: 'basic4',
> > +                    detail: 'light'
> > +                }
> > +            },
> > +            normalize: function(item) {
> > +                return {
> > +                    title: item.group_name + "/" + item.jar_name,
> > +                    subtitle: item.version,
> > +                    description: item.description,
>
> Seeing as item.description exists, you don't need to re-specify it here.
> We're actually extending the item obj with whatever you return here, so
> you don't lose any of the original properties from item
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/2087/files#r36799120>
> .
>